### PR TITLE
Remove pg_depend entries from columnar metadata indexes to columnar-am (inserted in #5456)

### LINKED ISSUE #5510: pg_dump gives columnar AM dependency loop warnings
When I run pg_dump against a database with Citus I get a series of warnings:
```sql
$ pg_dump
pg_dump: warning: could not resolve dependency loop among these items:
pg_dump:   ACCESS METHOD columnar  (ID 1769 OID 361215)
pg_dump:   INDEX stripe_pkey  (ID 3323 OID 361190)
pg_dump:   CONSTRAINT stripe_pkey  (ID 3324 OID 361191)                                  
pg_dump:   POST-DATA BOUNDARY  (ID 3482)    
pg_dump:   TABLE DATA test  (ID 3478 OID 361296)
pg_dump:   PRE-DATA BOUNDARY  (ID 3481)     
pg_dump: warning: could not resolve dependency loop among these items:
pg_dump:   ACCESS METHOD columnar  (ID 1769 OID 361215)                                  
pg_dump:   INDEX options_pkey  (ID 3319 OID 361185)
pg_dump:   CONSTRAINT options_pkey  (ID 3320 OID 361186)    
pg_dump:   POST-DATA BOUNDARY  (ID 3482)    
pg_dump:   TABLE DATA test  (ID 3478 OID 361296)
pg_dump:   PRE-DATA BOUNDARY  (ID 3481)
pg_dump: warning: could not resolve dependency loop among these items:
pg_dump:   ACCESS METHOD columnar  (ID 1769 OID 361215)
pg_dump:   INDEX chunk_group_pkey  (ID 3325 OID 361195)
pg_dump:   CONSTRAINT chunk_group_pkey  (ID 3326 OID 361196)
pg_dump:   POST-DATA BOUNDARY  (ID 3482)    
pg_dump:   TABLE DATA test  (ID 3478 OID 361296)
pg_dump:   PRE-DATA BOUNDARY  (ID 3481)                                                  
pg_dump: warning: could not resolve dependency loop among these items:                   
pg_dump:   ACCESS METHOD columnar  (ID 1769 OID 361215)
pg_dump:   INDEX chunk_pkey  (ID 3327 OID 361207)
pg_dump:   CONSTRAINT chunk_pkey  (ID 3328 OID 361208)
pg_dump:   POST-DATA BOUNDARY  (ID 3482)    
pg_dump:   TABLE DATA test  (ID 3478 OID 361296)
pg_dump:   PRE-DATA BOUNDARY  (ID 3481)     
pg_dump: warning: could not resolve dependency loop among these items:
pg_dump:   ACCESS METHOD columnar  (ID 1769 OID 361215)
pg_dump:   INDEX stripe_first_row_number_idx  (ID 3321 OID 361286)
pg_dump:   CONSTRAINT stripe_first_row_number_idx  (ID 3322 OID 361287)
pg_dump:   POST-DATA BOUNDARY  (ID 3482)                                                 
pg_dump:   TABLE DATA test  (ID 3478 OID 361296)
pg_dump:   PRE-DATA BOUNDARY  (ID 3481) 
```

fwiw, test is not a columnar table.

### PR DESCRIPTION
**Pull request by onurctirtir**
DESCRIPTION: Fixes (pg_dump/pg_upgrade) dependency loop warnings caused by pg_depend entries inserted by citus_columnar

Fixes #5510.

In the past, having columnar tables in the cluster was causing pg
upgrades to fail when attempting to access columnar metadata. This is
because, pg_dump doesn't see objects that we use for columnar-am related
booking as the dependencies of the tables using columnar-am.
To fix that; in #5456, we inserted some "normal dependency" edges (from
those objects to columnar-am) into pg_depend.

This helped us ensuring the existency of a class of metadata objects
--such as columnar.storageid_seq-- and helped fixing #5437.

However, the normal-dependency edges that we added for indexes on
columnar metadata tables --such columnar.stripe_pkey-- didn't help at
all because they were indeed causing dependency loops (#5510) and
pg_dump was not able to take those dependency edges into the account.

For this reason, this commit deletes those dependency edges so that
pg_dump stops complaining about them. Note that it's not critical to
delete those edges from pg_depend since they're not breaking pg upgrades
but were triggering some warning messages. And given that backporting
a sql change into older versions is hard a lot, we skip backporting
this.

### CODE REVIEW

File: src/test/regress/expected/multi_extension.out [hunk 1]
```diff
@@ -1250,6 +1250,44 @@ SELECT * FROM pg_dist_cleanup;
          2 |            0 |           1 | table_with_orphaned_shards_102011 |             0 |           0
 (2 rows)
 
+-- TODO: Looks like citus_columnar doesn't get auto-updated when updating Citus
```
Comment (onurctirtir) [inline src/test/regress/expected/multi_extension.out:1253]: ok, but at least open an issue for this

File: src/backend/columnar/sql/citus_columnar--11.1-1--11.2-1.sql [hunk 2]
```diff
@@ -1 +1,61 @@
 -- citus_columnar--11.1-1--11.2-1
+
+CREATE OR REPLACE FUNCTION columnar_internal.columnar_ensure_am_depends_catalog()
```
Comment (onurctirtir) [inline src/backend/columnar/sql/citus_columnar--11.1-1--11.2-1.sql:3]: do not define this here but fix the Makefile

File: src/backend/columnar/sql/udfs/columnar_ensure_am_depends_catalog/11.2-1.sql [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+         'pg_am'::regclass::oid as classid,
+         (select oid from pg_am where amname = 'columnar') as objid,
+         0 as objsubid,
+         -- ... to each object that is registered to pg_class and that lives
+         -- in "columnar" schema. That contains catalog tables, indexes
+         -- created on them and the sequences created in "columnar" schema.
+         --
+         -- Given the possibility of user might have created their own objects
+         -- in columnar schema, we explicitly specify list of objects that we
+         -- are interested in.
```
Comment (onurctirtir) [inline src/backend/columnar/sql/udfs/columnar_ensure_am_depends_catalog/11.2-1.sql:31]:
```suggestion
         -- ... to some objects registered as regclass and that lives in
         -- "columnar" schema. That contains catalog tables and the sequences
         -- created in "columnar" schema.
         --
         -- Given the possibility of user might have created their own objects
         -- in columnar schema, we explicitly specify list of objects that we
         -- are interested in.
```

File: src/backend/columnar/sql/udfs/columnar_ensure_am_depends_catalog/11.2-1.sql [hunk 4]
```diff
@@ -0,0 +1,43 @@
+CREATE OR REPLACE FUNCTION columnar_internal.columnar_ensure_am_depends_catalog()
```
[the remark this answers is not in the document]
Comment (onurctirtir) [reply src/backend/columnar/sql/udfs/columnar_ensure_am_depends_catalog/11.2-1.sql:1]: via citus_finish_pg_upgrade()

**Review by marcocitus — COMMENTED** [review]
Is it possible to have some kind of pg_dump test to confirm we fixed the original issue?

**Comment by onurctirtir** [comment]
> Is it possible to have some kind of pg_dump test to confirm we fixed the original issue?

I see that we have some pg_dump tests for columnar in pg_dump.sql but apparently they don't report warnings etc., so not sure if there is way to do that.

**Review by marcocitus — APPROVED** [review]

**Pull request merged**